x-pack/legacy/plugins/siem/public/components/header_page/editable_title.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import styled, { css } from 'styled-components';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFieldText,
+  EuiButtonIcon,
+} from '@elastic/eui';
+
+import { Title } from './title';
+
+const StyledEuiButtonIcon = styled(EuiButtonIcon)`
+  ${({ theme }) => css`
+    margin-left: ${theme.eui.euiSize};
+  `}
+`;
+
+StyledEuiButtonIcon.displayName = 'StyledEuiButtonIcon';
+
+interface Props {
+  submitTitle: string;
+  cancelTitle: string;
+  isLoading: boolean;
+  onClickEditIcon: () => void;
+  title: string | React.ReactNode;
+  editMode: boolean;
+  editIcon?: string;
+  onChange: (a: string) => void;
+  onCancel: () => void;
+  onSubmit: () => void;
+}
+
+const EditableTitleComponent: React.FC<Props> = ({
```

Comment:
The component should manage the state of showing or not showing the form inside of this component. We should not have to re-implement the same logic each time we want to use
.